xmtp_id/src/associations/signature.rs

```diff
@@ -182,24 +233,24 @@ impl Signature for InstallationKeySignature {
             signature: Some(SignatureKindProto::InstallationKey(
                 RecoverableEd25519SignatureProto {
                     bytes: self.bytes(),
+                    public_key: self.verifying_key.clone(),
                 },
             )),
         }
     }
 }
 
-#[allow(dead_code)]
 #[derive(Debug, Clone)]
 pub struct LegacyDelegatedSignature {
     // This would be the signature from the legacy key
     legacy_key_signature: RecoverableEcdsaSignature,
-    signed_public_key: xmtp_proto::xmtp::message_contents::SignedPublicKey,
+    signed_public_key: ValidatedLegacySignedPublicKey,
 }
 
 impl LegacyDelegatedSignature {
     pub fn new(
         legacy_key_signature: RecoverableEcdsaSignature,
-        signed_public_key: xmtp_proto::xmtp::message_contents::SignedPublicKey,
+        signed_public_key: ValidatedLegacySignedPublicKey,
```

Comment:
We can handle in a subsequent PR, but I do think it would be cleaner if these were unvalidated to start and all validation happened in `recover_signer` like the other signature types.